src/content/learn/thinking-in-react.md

```diff
@@ -492,6 +492,8 @@ Inside the `SearchBar`, you will add the `onChange` event handlers and set the p
   onChange={(e) => onFilterTextChange(e.target.value)} />
 ```
 
+(Set up changing state for the checkbox using `onInStockOnlyChange` yourself.)
+
 Now the application fully works!
```

Comment:
@smith558 The sentence here saying "Now the application fully works!" is misleading, since the application doesn't yet work, it needs further code added (unless the note I would like to add is there).

Reply:
It does work, as shown in the next sandbox, not the previous. The guide is written as "here's the code you'll right" and "here's a working sandbox with those changes".

Reply:
The sandbox works of course. I meant if you follow the steps in your own code editor. Look, it is inconsistent with the rest of the tutorial articles (as well as this one) where the pattern is that even the most obvious steps are explicitly mentioned. This is the only place in all of the tutorial articles that this pattern suddenly breaks for a moment. I just went through all of them in order.

Reply:
Decide as you wish but you've probably worked with these tutorials many many times and are well proficient with React as well. I tried this tutorial for the first time and it happened to be an issue. You can hardly have a better validation that a problem may exist for your target audience.

Reply:
@rickhanlonii would you please let me know if it seems to make more sense now?

Reply:
@smith558 i see, yeah I agree, thanks for explaining it for me more. 

Instead of this note, can we just expand the code above this to give a full example, highlighting the code to change like this:

<img width="855" alt="Screenshot 2024-01-22 at 5 00 04 PM" src="https://github.com/reactjs/react.dev/assets/2440089/e3ef12f8-57b8-4341-ae93-58d75415b2e1">

Here's the code:

```js
```js {4,5,13,19}
function SearchBar({
  filterText,
  inStockOnly,
  onFilterTextChange,
  onInStockOnlyChange
}) {
  return (
    <form>
      <input
        type="text"
        value={filterText}
        placeholder="Search..."
        onChange={(e) => onFilterTextChange(e.target.value)}
      />
      <label>
        <input
          type="checkbox"
          checked={inStockOnly}
          onChange={(e) => onInStockOnlyChange(e.target.checked)}
        />{" "}
        Only show products in stock
      </label>
    </form>
  );
}
```
```

Reply:
Done